Submitted By:            Armin K. <krejzi at email dot com>
Date:                    2014-07-23
Initial Package Version: 0.7.4
Upstream Status:         Unknown
Origin:                  Debian
Description:             Fixes building with latest libva

--- a/src/utils_glx.h	2012-10-05 17:02:58.000000000 +0200
+++ b/src/utils_glx.h	2014-07-23 14:00:20.967148712 +0200
@@ -48,6 +48,13 @@
 typedef void (*PFNGLXRELEASETEXIMAGEEXTPROC)(Display *, GLXDrawable, int);
 #endif
 
+#if GL_GLEXT_VERSION >= 85
+/* XXX: PFNGLMULTITEXCOORD2FPROC got out of the GL_VERSION_1_3_DEPRECATED
+   block and is not defined if GL_VERSION_1_3 is defined in <GL/gl.h>
+   Redefine the type here as an interim solution */
+typedef void (*PFNGLMULTITEXCOORD2FPROC) (GLenum target, GLfloat s, GLfloat t);
+#endif
+
 #ifndef GL_FRAMEBUFFER_BINDING
 #define GL_FRAMEBUFFER_BINDING GL_FRAMEBUFFER_BINDING_EXT
 #endif
--- a/src/vdpau_decode.c	2012-10-05 17:02:58.000000000 +0200
+++ b/src/vdpau_decode.c	2014-07-23 14:00:12.806040964 +0200
@@ -1289,7 +1289,7 @@
             driver_data,
             obj_context->vdp_decoder,
             obj_surface->vdp_surface,
-            (VdpPictureInfo)&obj_context->vdp_picture_info,
+            (VdpPictureInfo*)&obj_context->vdp_picture_info,
             obj_context->vdp_bitstream_buffers_count,
             obj_context->vdp_bitstream_buffers
         );
--- a/src/vdpau_dump.c	2012-10-05 17:02:58.000000000 +0200
+++ b/src/vdpau_dump.c	2014-07-23 14:00:15.886081172 +0200
@@ -59,8 +59,6 @@
         _(VAEncSequenceParameterBufferType);
         _(VAEncPictureParameterBufferType);
         _(VAEncSliceParameterBufferType);
-        _(VAEncH264VUIBufferType);
-        _(VAEncH264SEIBufferType);
 #endif
 #if VA_CHECK_VERSION(0,31,1)
         _(VAQMatrixBufferType);
